### PR TITLE
Add test for preset addition on expired mock. Abstract logic.

### PR DESCRIPTION
**Pull request by alexcouper**
@txels @IrynaMironava 

This makes some small changes following the #108 merge.

Once this has been merged it'll be worth a new release I think.

### CODE REVIEW

File: tests/integration/test_http.py [hunk 1]
```diff
@@ -394,3 +394,5 @@ def test_reply_depending_on_body():
 
     response, data = fake_client.post(url='/hello', body='345')
     assert_equals(response.status, 404)
+
+
```
Comment (txels) [inline tests/integration/test_http.py]: ?

**Comment by txels** [comment]
I just realised we have no Travis builds set up for this repo.

File: tests/integration/test_preset.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    get a 404.
+    """
+
+    http_mock = HTTPMock('localhost', 8000, timeout=0.1)
+    time.sleep(0.3)
+    preset = http_mock.when('POST /fred/test/one')
+    assert_raises(
+        ConfigurationError, preset.reply, b'You tested fred well', 200
+    )
+
```
Comment (txels) [inline tests/integration/test_preset.py]: pep8 blank line...

**Comment by txels** [comment]
We need a proper build with flake8 etc.

**Comment by alexcouper** [comment]
I'll create another PR that fixes a bunch of pep8 stuff...

**Comment by txels** [comment]
@alexcouper https://landscape.io/github/pretenders/pretenders/4

**Pull request merged**